Sub/Src/Mission/waypoint_task.py

```diff
@@ -0,0 +1,165 @@
+'''
+Copyright 2019, David Pierce Walker-Howell, All rights reserved
+Author: David Pierce Walker-Howell<piercedhowell@gmail.com>
+Last Modified 07/3/2019
+Description: This module defines how to perform a waypoint task. It
+            ingest's a dictionary containing all of the necessary parameters
+            to perform a waypoint mission. This is passed in from the Mission
+            Controller.
+'''
+import sys
+import os
+import time
+
+HELPER_PATH = os.path.join("..", "Helpers")
+sys.path.append(HELPER_PATH)
+import util_timer
+
+PARAM_PATH = os.path.join("..", "Params")
+sys.path.append(PARAM_PATH)
+MECHOS_CONFIG_FILE_PATH = os.path.join(PARAM_PATH, "mechos_network_configs.txt")
+from mechos_network_configs import MechOS_Network_Configs
+from MechOS import mechos
+import csv
+import numpy as np
+
+class Waypoint_Task:
+    '''
+    Given the waypoint task description dictionary, drive along the waypointed
+    route. If the the task cannot be completed within the given timeout time,
+    exit the mission and considerate it a failure.
+    '''
+    def __init__(self, waypoint_task_dict, drive_functions):
+        '''
+        Initialize the waypoint task given the waypoint dict.
+
+        Parameters:
+            waypoint_task_dict: A python dictionary containing the parameters
+                                for the waitpoint mission.
+                            Dictionary Form:
+                            -------------------
+                            {
+                                "type": "Waypoint"
+                                "name": <task_name>
+                                "timeout": <timeout_time_minutes>
+                                "buffer_zone": <buffer zone for each (North, East) position (ft)>
+                                "waypoint_file": <location of csv file containing waypoints>
+                            }
+            drive_functions: An already initialized object of drive_functions. This
+                            class contains a variety of drive functions that are useful
+                            to complete tasks.
+        '''
+        self.waypoint_task_dict = waypoint_task_dict
+        self.name = self.waypoint_task_dict["name"]
+        self.type = "Waypoint"
+
+        self.drive_functions = drive_functions
+
+        #Initialize the timeout timer
+        self.timeout_timer = util_timer.Timer()
+
+        self.print_task_info()
```

Comment:
Remove the print task information and move it to the beginning of the run function.